roachtest: This commit guards against a nil dereference

smg260 opened the pull request:
of a fileless child logger's file name, which can happen due to a race condition in which we attempt to log whilst the test is being torn down.

See also: https://github.com/cockroachdb/cockroach/pull/81779

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/92845)
<!-- Reviewable:end -->

Commit:
roachtest: This commit guards against a nil dereference
of a child logger's file, which can happen due to a race
condition in which we attempt to log whilst the test is
being torn down.

See also: https://github.com/cockroachdb/cockroach/pull/81779
Release note: None
Epic: None

smg260 commented:
bores r+

smg260 commented:
bors r+

renatolabs commented:
Help me understand this change: if the issue happens due to a race condition, wouldn't it still exist even after this change (TOCTTOU)?
